File: backend/app/schemas/learning.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class LearningSessionCreate(BaseModel):
    topic_id: str = Field(..., description="Topic ID to start learning")


class LearningSessionUpdate(BaseModel):
    completed: Optional[bool] = Field(None, description="Mark session as completed")


class LearningSessionResponse(BaseModel):
    id: str
    student_id: str
    topic_id: str
    start_time: datetime
    end_time: Optional[datetime] = None
    duration_minutes: Optional[int] = None
    completed: bool = False
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class LearningSessionStats(BaseModel):
    total_sessions: int = Field(..., description="Total number of learning sessions")
    total_time_minutes: int = Field(..., description="Total time spent learning (minutes)")
    sessions_this_week: int = Field(..., description="Sessions in the last 7 days")
    time_this_week_minutes: int = Field(..., description="Time spent this week (minutes)")
    unique_topics_learned: int = Field(..., description="Number of unique topics completed")

File: frontend/src/lib/api/learning.ts
import apiClient from './client'

export interface LearningSession {
  id: string
  student_id: string
  topic_id: string
  start_time: string
  end_time?: string
  duration_minutes?: number
  completed: boolean
  created_at: string
  updated_at: string
}

export interface LearningSessionStats {
  total_sessions: number
  total_time_minutes: number
  sessions_this_week: number
  time_this_week_minutes: number
  unique_topics_learned: number
}

export const learningApi = {
  // Start a new learning session
  startSession: async (topicId: string): Promise<LearningSession> => {
    return apiClient.post('/api/v1/learning/sessions', {
      topic_id: topicId,
    })
  },

  // End a learning session
  endSession: async (
    sessionId: string,
    completed: boolean = false
  ): Promise<LearningSession> => {
    return apiClient.patch(`/api/v1/learning/sessions/${sessionId}`, {
      completed,
    })
  },

  // Get learning sessions
  getSessions: async (topicId?: string, limit: number = 20): Promise<LearningSession[]> => {
    const params = new URLSearchParams()
    if (topicId) params.append('topic_id', topicId)
    params.append('limit', limit.toString())

    return apiClient.get(`/api/v1/learning/sessions?${params.toString()}`)
  },

  // Get a specific session
  getSession: async (sessionId: string): Promise<LearningSession> => {
    return apiClient.get(`/api/v1/learning/sessions/${sessionId}`)
  },

  // Get learning statistics
  getStats: async (): Promise<LearningSessionStats> => {
    return apiClient.get('/api/v1/learning/sessions/stats')
  },
}
